clusterloader2/testing/watch-list/config.yaml

```diff
@@ -0,0 +1,84 @@
+name: huge-secrets
+tuningSets:
+- name: Uniform10qps
+  qpsLoad:
+    qps: 10
+steps:
+- name: Create well known namespaces
+  phases:
+  # don't know why, but it has to be set
+  # otherwise I get some compilation errors
+  - namespaceRange: null
+    # this seems to be controlling the
+    # number of ns I want to create
+    replicasPerNamespace: 1
+    tuningSet: Uniform10qps
+    objectBundle:
+    - basename: huge-secrets
+      objectTemplatePath: "ns.yaml"
+- name: Create secrets
+  phases:
+  - namespaceRange:
+      min: 0
+      max: 0
+      basename: huge-secrets
+    replicasPerNamespace: 400
+    tuningSet: Uniform10qps
+    objectBundle:
+    - basename: huge-secret
+      objectTemplatePath: "secret.yaml"
+- name: Start measurements
+  measurements:
+    - Identifier: MemoryProfile
+      Method: MemoryProfile
+      Params:
+        action: start
+    - Identifier: WaitForControlledPodsRunning
+      Method: WaitForControlledPodsRunning
+      Params:
+        action: start
+        apiVersion: apps/v1
+        kind: Deployment
+        labelSelector: group = watch-list
+        operationTimeout: 60s
+- name: Start the secret informer
+  phases:
+    - namespaceRange:
+        min: 0
+        max: 0
+        basename: huge-secrets
+      replicasPerNamespace: 1
+      tuningSet: Uniform10qps
+      objectBundle:
+        - basename: huge-secret
+          objectTemplatePath: role.yaml
+        - basename: huge-secret
+          objectTemplatePath: roleBinding.yaml
+        - basename: huge-secret
+          objectTemplatePath: "deployment.yaml"
+          templateFillMap:
+            Replicas: 1
+- name: Wait for the secret informer pod to be running
+  measurements:
+    - Identifier: WaitForControlledPodsRunning
+      Method: WaitForControlledPodsRunning
+      Params:
+        action: gather
+- name: Measure the API server memory usage
```

Comment:
this is untested because `kind` (my local env) is not supported.

Reply:
Yes, this will collect memory profiles, but I don't think it will gather number of bytes (maybe you can infer this somehow from pprofs).

You may want to use TestMetrics (https://github.com/kubernetes/perf-tests/blob/4da7097e35b13225ed05bf3dfcb842864c11fae4/clusterloader2/testing/access-tokens/config.yaml#L60-L67)

which gathers quite a lot of metrics (including cpu/memory absolute values)/pprofs and other stuff that may be useful.

Reply:
mhm, ok, is there a way to turn profiles into a dashboard so that it easier to compare/visualise?

Reply:
oh, it looks great!

Reply:
do you know how memory-related metrics are gathered from the api server?

Reply:
okay, it looks like by default (no params defined) it will bring memory stats :)